Sources/SwiftParser/Modifiers.swift

```diff
@@ -167,4 +148,63 @@ extension Parser {
 
     return RawDeclModifierSyntax(name: keyword, detail: detail, arena: self.arena)
   }
+
+  mutating func parseAccessLevelModifier() -> RawDeclModifierSyntax {
+    assert(
+      [
+        RawTokenKind.privateKeyword,
+        .fileprivateKeyword,
+        .internalKeyword,
+        .publicKeyword
+      ].contains(self.currentToken.tokenKind)
+    )
+    let name = self.consumeAnyToken()
+    let details: RawDeclModifierDetailSyntax?
+    if let lparen = self.consume(if: .leftParen) {
+      var beforeDetail: [RawTokenSyntax] = []
+      var detail: RawTokenSyntax? = nil
+      while !atAny([.eof, .rightParen]) {
+        if currentToken.isContextualKeyword("set") {
+          detail = consume(remapping: .contextualKeyword)
+          break
+        }
+        beforeDetail.append(consumeAnyToken())
+      }
+      var afterDetail: [RawTokenSyntax] = []
+      if let _ = detail {
+        while !atAny([.eof, .rightParen]) {
+          afterDetail.append(consumeAnyToken())
+        }
+      } else {
+        afterDetail = beforeDetail
+        beforeDetail = []
+      }
+      let rparen = consume(if: .rightParen) ??
```

Comment:
Is `expect` not sufficient to recover to the right paren? I would expect commas wouldn't bind as tightly as parentheses here so you wouldn't have to do the extra token munging here.

Reply:
@CodaFi

I made single `expect(.rightParen)` version here.
Is this match with your recommendation?

https://github.com/omochi/swift-syntax/pull/1/files

This implementation is simpler.
And safer when there is no stopper token. 
For example:

```swift
private(
#endif
```

`expect` stops to scanning at `#endif` without eating many tokens following from here.

But diagnostics for 

```swift
private(get, set, didSet)
```

is worse than before.
(see diff https://github.com/omochi/swift-syntax/pull/1/files#diff-ad19f5e399c23c3b244b7e592ced5c7b7013e224463b61962f131d4fbec70599R159-R161)

The reason of changing diagnostics is comma and right-paren have same `TokenPrecedence` level.
`expect` stops to search right-paren when see comma.
How do you think about this change of diagnostics?

If you say that previous diagnostics is better,
I try to achieve both using `expect` and better diagnostics.
I could be it with `expectAny([.rightParen, .comma)]` and some loops.